server/pages/index.js

```diff
@@ -128,6 +128,13 @@ function getDefaultContext( request ) {
 		initialServerState = getInitialServerState( serializeCachedServerState );
 	}
 
+	const protocol = request.get( 'X-Forwarded-Proto' ) === 'https' ? 'https' : 'http';
```

Comment:
Like I mentioned in my earlier comment, I thought about using properties of the request to build up the URL.  However, I saw how the login redirect URL was being built up in the setUpLoggedInRoute.  Specifically, right here: https://github.com/Automattic/wp-calypso/blob/87f4b5c35f6d7f56a01c292a7bdc2534d4b72a26/server/pages/index.js#L211

I thought that it'd be better to use the same kinds of method to build up the redirect URI here.